### PR TITLE
WIP: Add inline image resizing UI

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

This PR adds UI to resize inline images. This is a more hidden feature. This PR should aim to add the minimum UI needed so that the image dimensions can be changed without TinyMCE's resize handles.

* Should `PostitionedAtSelection` be moved to the `rich-text` package or the `components` package?

screenshot 2019-02-07 at 16 49 00

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by iseulde** [comment]
@jasmussen I might need your help with the design of this. :) Ideally it should look a bit like the link input pop up. Should the input have a label? It's otherwise unclear what the input field is for? Feel free to change anything on this PR!

**Comment by jasmussen** [comment]
Cool!

I pushed a little polish:

screenshot 2019-02-08 at 09 52 44

Is there a chance we can get some resize handles? Or is this the new interface?

**Comment by iseulde** [comment]
@jasmussen Thanks! The image block has resize handles that we added. Currently the _inline_ images have resize handles that TinyMCE added. I'm trying to add some minimal accessible UI to the inline images when we pull out TinyMCE, so resizing is still possible. As this is a lesser used feature, adding resize handles is not a top priority.

**Comment by jasmussen** [comment]
Works for me. Shine on!

**Comment by iseulde** [comment]
Rebased. Seems to work fine now.

There's a small issue with the image format that is more obvious after this PR. The image object receives boundaries, but it shouldn't, as it's just an object, and at those boundaries, the UI is displayed. This issue as already present in `master`. I plan to fix it with a refactoring of the object format (don't worry, no API changes). The idea is that the objects are so different from formats that it justifies a separate key in the rich text value.

* There can only be one per index.
* We have to add `object: true` to it for special handling.
* It should not have format boundaries.
* I think there are some problems during serialisation where the object should always be last, but after applying a format it isn't.

With a separate key for objects, all these issues would go away. Perhaps something similar should be done for line separator formats (read: lists).

**Comment by youknowriad** [comment]
Some issues:

 - When I enter the format from the left (keyboard), the width input is empty
 - Why do I have to type "Enter" to apply the change, since the input is directly visible, I expect that the change is applied without submitting the form.
 - Copy pasting text with an inline image, results in a paragraph block and an image block.

**Comment by iseulde** [comment]
> * Why do I have to type "Enter" to apply the change, since the input is directly visible, I expect that the change is applied without submitting the form.

It's like the link input. Without enter, the image would continuously change in size.

**Comment by iseulde** [comment]
> * Copy pasting text with an inline image, results in a paragraph block and an image block.

This is an issue in `master` as well. Pasted images rarely occur outside of paragraphs, so they are taken out.

**Comment by iseulde** [comment]
> * When I enter the format from the left (keyboard), the width input is empty

I thought this issue would be gone after f96dbe0. Could reproduce before but not after.

File: packages/format-library/src/image/index.js [hunk 1]
```diff
@@ -73,6 +113,50 @@ export const image = {
...
 						} }
 					/> }
+					{ isActive && <PositionedAtSelection key={ key }>
+						<Popover
+							position="bottom center"
+							focusOnMount={ false }
+						>
+							{ // Disable reason: KeyPress must be suppressed so the block doesn't hide the toolbar
+							/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */ }
+							<form
```
Comment (noisysocks) [inline packages/format-library/src/image/index.js]: Should we move some of this markup into its own component? `ImageEdit` is getting rather large.

**Review by noisysocks — APPROVED** [review]
This tests well and is looking pretty good!

Some possible improvements:

1. It's strange that the Apply button is not the same height as the `<input>` nor the popover:

   screen shot 2019-02-18 at 11 04 50

2. When you first insert an inline image using the inserter, the popover is pointing to the right of the image instead of the centre:

   screen shot 2019-02-18 at 11 14 26

3. The `PositionAtSelection` component needs a `README.md` and possibly some unit tests.

I'd be happy to include this as is in the GB 5.1 RC. That way, we're not regressing any functionality by shipping GB 5.0 with resize handles and GB 5.1 with none. We can iterate as part of GB 5.2.

File: packages/components/src/positioned-at-selection/index.js [hunk 2]
```diff
@@ -63,5 +63,3 @@ class PositionedAtSelection extends Component {
 		);
 	}
 }
-
-export default PositionedAtSelection;
```
[the remark this answers is not in the document]
Comment (noisysocks) [reply packages/components/src/positioned-at-selection/index.js]: d03c77f

[the remark this answers is not in the document]
Comment (iseulde) [reply packages/components/src/positioned-at-selection/index.js]: Thanks!

**Comment by iseulde** [comment]
Thanks @noisysocks! Cc @jasmussen for the design. I think the pop up should look closer to the one we have for links.

**Pull request merged**